src/backend/marsha/core/tests/test_api_liveregistration.py

```diff
@@ -0,0 +1,790 @@
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+        jwt_token.payload["roles"] = [random.choice(["administrator", "instructor"])]
+
+        response = self.client.post(
+            "/api/liveregistrations/",
+            {"email": "salome@test-fun-mooc.fr", "should_send_reminders": True},
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+
+        self.assertEqual(response.status_code, 201)
+        created_liveregistration = LiveRegistration.objects.last()
+        self.assertEqual(
+            json.loads(response.content),
+            {
+                "email": "salome@test-fun-mooc.fr",
+                "id": str(created_liveregistration.id),
+                "lti_consumer_site": None,
+                "lti_user_id": None,
+                "should_send_reminders": True,
+                "video": str(video.id),
+            },
+        )
+
+    def test_api_liveregistration_create_cant_register_when_not_scheduled(self):
+        """Can't register if video is not scheduled."""
+        video = VideoFactory()
+        self.assertFalse(video.is_scheduled)
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+
+        response = self.client.post(
+            "/api/liveregistrations/",
+            {"email": "salome@test-fun-mooc.fr", "should_send_reminders": True},
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(
+            json.loads(response.content),
+            {"reason": f"video with id {str(video.id)} doesn't accept registration."},
+        )
+
+    def test_api_liveregistration_create_cant_register_when_no_email(self):
+        """Can't register to a scheduled video if there is no email."""
+        starting_at = timezone.now() + timedelta(days=5)
+        video = VideoFactory(starting_at=starting_at)
+        self.assertTrue(video.is_scheduled)
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+
+        response = self.client.post(
+            "/api/liveregistrations/",
+            {"should_send_reminders": True},
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(
+            json.loads(response.content),
+            {"email": ["This field is required."]},
+        )
+
+    def test_api_liveregistration_create_cant_register_two_identical_emails(self):
+        """Same email can't be used for the same video."""
+        starting_at = timezone.now() + timedelta(days=5)
+        video = VideoFactory(starting_at=starting_at)
+        LiveRegistrationFactory(email="salome@test-fun-mooc.fr", video=video)
+        self.assertTrue(video.is_scheduled)
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+
+        with self.assertRaises(IntegrityError):
```

Comment:
You should never expect an `IntegrityError` here. In production this mean you will a `500` error in sentry. This error must be catch and managed in your code

Reply:
got it !